### PR TITLE
instructor session results, 'by question view': add the ability to view the profile photo #3028

### LINKED ISSUE #3028: instructor session results, 'by question view': add the ability to view the profile photo
Currently, in other views (e.g. 'by giver'), the profile photo can be viewed by hovering over the name.

### PR DESCRIPTION
**Pull request by JosephineKwa**
Fixes #3028

### CODE REVIEW

**Comment by damithc** [comment]
@thyageshm for your review please

**Comment by JosephineKwa** [comment]
Here are the changes that were made:

When the user hovers over the giver or recipient name, the "View Photo" option appears.



The photo is displayed when the "View Photo" option is clicked.



The "View Photo" option doesn't appear for teams or instructors, it only appears when the mouse is hovered over student's names.

File: src/main/webapp/jsp/instructorFeedbackResultsByQuestion.jsp [hunk 1]
```diff
@@ -215,9 +252,27 @@
...
+                                                                <%} else {%>
+                                                                <%=data.bundle.getFullNameFromRoster(prevGiver)%>
+                                                                <%}%>                                                            
+                                                            </td>
                                                             <td class="middlealign color_neutral"><%=data.bundle.getTeamNameFromRoster(prevGiver)%></td>
-                                                            <td class="middlealign color_neutral"><%=data.bundle.getFullNameFromRoster(possibleReceiver)%></td>
+                                                            <td class="middlealign color_neutral">
+                                                                <%if (question.recipientType == FeedbackParticipantType.SELF || question.recipientType == FeedbackParticipantType.STUDENTS || question.recipientType == FeedbackParticipantType.OWN_TEAM_MEMBERS || question.recipientType == FeedbackParticipantType.OWN_TEAM_MEMBERS_INCLUDING_SELF) {%>
```
Comment (thyageshm) [inline src/main/webapp/jsp/instructorFeedbackResultsByQuestion.jsp]: hey! you could consider extracting this check into a boolean function in the question object. smth like: isRecipientAStudent()

File: src/test/java/teammates/test/pageobjects/InstructorFeedbackResultsPage.java [hunk 2]
```diff
@@ -332,6 +332,58 @@ public void hoverAndViewStudentPhotoOnBody(int panelBodyIndex, String urlRegex)
...
+        
+        AssertHelper.assertContainsRegex(urlRegex, 
+                browser.driver.findElements(By.cssSelector(".popover-content > img"))
+                              .get(0)
+                              .getAttribute("src"));
+        
+        jsExecutor.executeScript("document.getElementsByClassName('popover')[0].parentNode.removeChild(document.getElementsByClassName('popover')[0])");
+    }
+    
+    public void hoverClickAndViewRecipientPhotoOnTableCell(int questionBodyIndex, int tableRow, String urlRegex) throws Exception {
```
Comment (thyageshm) [inline src/test/java/teammates/test/pageobjects/InstructorFeedbackResultsPage.java]: any specific reason we have a separate function for this? I would suggest creating a generic function that takes in the column number as a parameter, then both these functions can call that one with the appropriate column number

**Comment by thyageshm** [comment]
@JosephineKwa a very good job done. A few minor tweaks and I think this one is good to go!

**Comment by JosephineKwa** [comment]
Hey @thyageshm, sorry about the delay! Regarding the functions hoverClickAndViewRecipientPhotoOnTableCell and hoverClickAndViewGiverPhotoOnTableCell, I initially decided to separate them as not all the columns in the table contain students names. Some of them contain feedback or the team's name, thus I wanted to limit it to just hovering over the Giver and Recipient column. 

However looking at it now, I see that there is indeed a lot of duplicate code between the two functions, and I have made the necessary changes based on your suggestions above.

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackResultsPageUiTest.java [hunk 3]
```diff
@@ -215,6 +215,12 @@ public void testViewPhotoAndAjaxForLargeScaledSession() throws Exception {
        
         resultsPage.verifyHtmlAjax("/instructorFeedbackResultsAjaxByQuestion.html");
         
+        ______TS("test view photo for view by questions");
+        
+        resultsPage.removeNavBar();
+        resultsPage.hoverClickAndViewPhotoOnTableCell(0, 0, 0, "studentProfilePic?studentemail={*}&courseid={*}&user=CFResultsUiT.instr");
+        resultsPage.hoverClickAndViewPhotoOnTableCell(0, 0, 2, "profile_picture_default.png");
```
Comment (thyageshm) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackResultsPageUiTest.java]: Hi! I believe I wasnt completely clear on my intensions haha. Combining the functions into one was well done, but keep these as they were (ie hoverClickAndViewGiverPhotoOnTableCell and hoverClickAndViewRecipientPhotoOnTableCell) and in those functions call the new function you have created. That way when someone reads your test cases they understand why you have 2 hover-and-click tests, while at the same time your code is not replicated

Comment (thyageshm) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackResultsPageUiTest.java]: Once you are done with this, just make sure dev is green and I will merge this one in :)

**Comment by JosephineKwa** [comment]
Hi @thyageshm, I have modified the code based on your suggestions and gotten dev green.

**Comment by thyageshm** [comment]
Sorry, will merge this in today

**Pull request merged**